Marlin/Marlin_main.cpp

```diff
@@ -1613,9 +1613,20 @@ inline float set_homing_bump_feedrate(AxisEnum axis) {
 inline void line_to_current_position() {
   planner.buffer_line(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS], MMM_TO_MMS(feedrate_mm_m), active_extruder);
 }
+
 inline void line_to_z(float zPosition) {
```

Comment:
Note that `line_to_z(z)` does not update `current_position[Z_AXIS]`. It's called only in `run_z_probe` for the probing moves (where it doesn't need to update `current_position`). The `current_position` is set from `stepper.get_axis_position_mm(Z_AXIS)` at the end.